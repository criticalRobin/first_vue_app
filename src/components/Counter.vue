<template>
  <h2>Counter</h2>
  <p>{{ counter }} <sup>2</sup> = {{ getSquareValue }}</p>
</template>

<script>
export default {
  name: "Counter",
  data() {
    return {
      counter: 5,
    };
  },
  // similares a methods, pero computed se quedan en la cache
  computed: {
    getSquareValue() {
      return Math.pow(this.counter, 2);
    },
  },
};
</script>

<style></style>
