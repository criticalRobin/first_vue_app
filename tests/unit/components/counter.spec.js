import Counter from "@/components/Counter.vue";
import { shallowMount, mount } from "@vue/test-utils";

// describe es una función global de Jest que agrupa pruebas relacionadas
describe("Pruebas del Componente Counter", () => {
  // test es otra función global de Jest que permite definir una prueba
  test("Debe hacer match con el snapshot", () => {
    // shallowMount es una función de Vue Test Utils que monta un componente de manera superficial
    // Esto significa que no se montan los componentes hijos
    const wrapper = shallowMount(Counter);

    // expect es otra función global de Jest que permite hacer afirmaciones
    // .toMatchSnapshot() es un matcher de Jest que compara la salida actual con un snapshot previo
    // Un snapshot es una representación serializada del output de una prueba que se guarda en un archivo
    // En pruebas futuras, Jest compara la salida con el snapshot guardado
    // Si la salida coincide con el snapshot, la prueba pasa. Si no, la prueba falla
    expect(wrapper.html()).toMatchSnapshot();
  });
});
