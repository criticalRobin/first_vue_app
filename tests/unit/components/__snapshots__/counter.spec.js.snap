// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del Componente Counter Debe hacer match con el snapshot 1`] = `
<h2>Contador</h2>
<p>0 <sup>2</sup> = 0</p>
<div class="buttons"><button>+1</button><button>-1</button></div>
`;
